### PR TITLE
adding my keymap and user space

### PR DESCRIPTION
**Pull request by trudkin**
## Description

Code to have for my three basic keyboards.  Kept in unison by userspace code.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* n/a

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by trudkin** [comment]
Think I have made all the requested changes.  I see that it is failed in CI
on code I have not committed.   Is there anything I need to do at this time?

**Review by drashna — APPROVED** [review]

File: quantum/process_keycode/process_combo.h [hunk 1]
```diff
@@ -48,7 +48,7 @@ typedef struct {
 
 #define COMBO_END 0
 #ifndef COMBO_COUNT
-#    define COMBO_COUNT 0
+#    define COMBO_COUNT 8
```
Comment (zvecr) [inline quantum/process_keycode/process_combo.h:51]:
```suggestion
#    define COMBO_COUNT 0
```

File: tmk_core/common/keycode.h [hunk 2]
```diff
@@ -16,7 +16,7 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
 
 /*
- * Keycodes based on HID Keyboard/Keypad Usage Page (0x07) plus media keys from Generic Desktop Page (0x01) and Consumer Page (0x0C)
+ * Keycodes based on HID Keyboard/Keypad U/sage Page (0x07) plus media keys from Generic Desktop Page (0x01) and Consumer Page (0x0C)
```
Comment (zvecr) [inline tmk_core/common/keycode.h:19]:
```suggestion
 * Keycodes based on HID Keyboard/Keypad Usage Page (0x07) plus media keys from Generic Desktop Page (0x01) and Consumer Page (0x0C)
```

**Review by zvecr — CHANGES_REQUESTED** [review]

**Comment by trudkin** [comment]
Is there anything I need toget thir pushed?

File: keyboards/preonic/keymaps/trudkin/mykb.json [hunk 3]
```diff
@@ -0,0 +1 @@
...Y", "COLEMAK", "DVORAK", "KC_NO", "KC_PSCR", "KC_NLCK", "RGB_SPD", "RGB_MOD", "RGB_RMOD", "RGB_TOG", "KC_MAKE", "RESET", "DEBUG", "KC_TRNS", "KC_TRNS", "TERM_ON", "TERM_OFF", "RGB_VAI", "RGB_VAD", "RGB_M_R", "RGB_M_X", "RGB_M_B", "KC_TRNS", "KC_TRNS", "MU_MOD", "AU_ON", "AU_OFF", "AG_NORM", "AG_SWAP", "RGB_HUI", "RGB_HUD", "RGB_M_K", "RGB_M_SW", "RGB_M_SN", "KC_TRNS", "MUV_DE", "MUV_IN", "MU_ON", "MU_OFF", "MI_ON", "MI_OFF", "RGB_SAI", "RGB_SAD", "RGB_M_T", "RGB_M_X", "RGB_M_G", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS", "KC_TRNS"], ["KC_TRNS", "_________________NUMPAD_1__________________", "_________________NUMPAD_1__________________", "KC_TRNS", "KC_TRNS", "_________________NUMPAD_2__________________", "_________________NUMPAD_2__________________", "KC_TRNS", "KC_TRNS", "_________________NUMPAD_3__________________", "_________________NUMPAD_3__________________", "KC_TRNS", "KC_TRNS", "KC_TRNS", "_______NUMPAD_4________", "KC_TRNS", "KC_TRNS", "_______NUMPAD_4________", "KC_TRNS", "KC_TRNS", "KC_NO", "NUMPAD", "KC_NO", "KC_NO", "KC_TRNS", "KC_NO", "KC_NO", "KC_TRNS", "KC_NO", "KC_NO", "KC_NO", "KC_NO"]]}
```
Comment (noroadsleft) [inline keyboards/preonic/keymaps/trudkin/mykb.json:1]: This file should be removed; it's not going to work with any of QMK's tooling.

`git rm --cached -- keyboards/preonic/keymaps/trudkin/mykb.json` to remove it from Git while leaving it present on your system. Omit the `--cached` argument if you want to remove it from Git *and* your system.

**Review by noroadsleft — CHANGES_REQUESTED** [review]
drashna's initial review also shows two issues that are not resolved, in addition to my comment below.

- https://github.com/qmk/qmk_firmware/pull/11221#discussion_r547574874
- https://github.com/qmk/qmk_firmware/pull/11221#discussion_r547575622

File: keyboards/preonic/keymaps/trudkin/mykb.json (hunk 3 above)
Comment (trudkin) [reply keyboards/preonic/keymaps/trudkin/mykb.json:1]: changes  made

Comment (noroadsleft) [reply keyboards/preonic/keymaps/trudkin/mykb.json:1]: I'm not seeing this change on the pull request. You probably need to merge your GitHub fork's `master` branch with your local (on your computer) `master` branch, then run:

```
git commit -m "Delete keyboards/preonic/keymaps/trudkin/mykb.json"
git push
```

**Comment by noroadsleft** [comment]
Issue at https://github.com/qmk/qmk_firmware/pull/11221#discussion_r547575622 still needs resolution.

**Comment by trudkin** [comment]
[image: image.png]


Not sure why this is in this state, but the code is resolved.

TR

**Comment by noroadsleft** [comment]
Forgot to `git push` from your local repository, maybe?

**Comment by trudkin** [comment]
I pushed the file, I hope I did  it right as git is new to me.

File: compile_commands.json [hunk 4]
```diff
@@ -0,0 +1,167 @@
+[
```
Comment (drashna) [inline compile_commands.json:1]: Could you remove this file?

File: keyboards/preonic/keymaps/trudkin/config.h [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+  * but WITHOUT ANY WARRANTY; without even the implied warranty of 
+  * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the 
+  * GNU General Public License for more details. 
+  * 
+  * You should have received a copy of the GNU General Public License 
+  * along with this program.  If not, see <http://www.gnu.org/licenses/>. 
+  */ 
+#pragma once
+
+#include "../../config.h"
```
Comment (drashna) [inline keyboards/preonic/keymaps/trudkin/config.h:18]: This isn't needed. 
```suggestion
```

File: users/trudkin/adv_key.c [hunk 6]
```diff
@@ -0,0 +1,112 @@
+
```
Comment (drashna) [inline users/trudkin/adv_key.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: users/drashna/drashna.h [hunk 7]
```diff
@@ -48,7 +48,7 @@ enum userspace_layers {
     _MALTRON,
     _EUCALYN,
     _CARPLAX,
-    _MODS, /* layer 8 */
+    _THUMB, /* layer 8 */
```
Comment (drashna) [inline users/drashna/drashna.h:51]: Could you revert this?

**Review by drashna — CHANGES_REQUESTED** [review]

File: compile_commands.json (hunk 4 above)
Comment (trudkin) [reply compile_commands.json]: Done

File: keyboards/preonic/keymaps/trudkin/mykb.json (hunk 3 above)
Comment (trudkin) [reply keyboards/preonic/keymaps/trudkin/mykb.json]: This should be correct

File: keyboards/preonic/keymaps/trudkin/config.h [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+#define AUDIO_CLICKY
+#define MUSIC_MASK (keycode != KC_NO)
+
+/*
+ * MIDI options
+ */
+
+/* Prevent use of disabled MIDI features in the keymap */
+//#define MIDI_ENABLE_STRICT 1
+
```
Comment (fauxpark) [inline keyboards/preonic/keymaps/trudkin/config.h:69]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by trudkin — COMMENTED** [review]
made update

File: keyboards/ergodox_ez/keymaps/trudkin/keymap.c [hunk 9]
```diff
@@ -0,0 +1,519 @@
...
+            }
+
+            break;
+  }
+  return true;
+}
+
+
+
+uint32_t layer_state_set_user(uint32_t state) {
```
Comment (drashna) [inline keyboards/ergodox_ez/keymaps/trudkin/keymap.c:460]:
```suggestion
layer_state_t layer_state_set_user(layer_state_t state) {
```

**Comment by trudkin** [comment]
Lot of  changes on both code bases, will try again once my base is stable

**Pull request closed without merging**